File: ios/RCTImageMarker/TextBackground.m
//
//  NSObject+TextBackground.c
//  RCTImageMarker
//
//  Created by Jimmydaddy on 2023/6/22.
//  Copyright © 2023 Jimmy. All rights reserved.
//

#include "TextBackground.h"
#include "RCTConvert+ImageMarker.h"
#import <React/RCTBridge.h>
#import <React/RCTConvert.h>
#import "Utils.h"

@implementation TextBackground

-(id)initWithTextBackgroundStyle:(NSDictionary *) textBackground
{
    if (textBackground == nil) return nil;
    _typeBg = textBackground[@"type"];
    _paddingX = [RCTConvert CGFloat: textBackground[@"paddingX"]];
    _paddingY = [RCTConvert CGFloat: textBackground[@"paddingY"]];
    if([textBackground[@"color"] length] > 1) {
        _colorBg = [Utils getColor:textBackground[@"color"]];
    } else {
        _colorBg = [UIColor clearColor];
    }
    return self;
    
}
@end

File: ios/RCTImageMarker/TextBackground.swift
//
//  TextBackground.swift
//  RCTImageMarker
//
//  Created by Jimmydaddy on 2023/6/22.
//  Copyright © 2023 Jimmy. All rights reserved.
//

import Foundation
import UIKit
import React

class TextBackground: Padding {
    var typeBg: String?
    var colorBg: UIColor?
    var cornerRadius: CornerRadius?

    init?(textBackgroundStyle textBackground: [AnyHashable: Any]?) throws {
        guard let textBackground = textBackground, !Utils.isNULL(textBackground) else {
            return nil
        }
        try super.init(paddingData: textBackground)
        self.typeBg = textBackground["type"] as? String
        guard let color = textBackground["color"] as? String else {
            throw NSError(domain: ErrorDomainEnum.PARAMS_REQUIRED.rawValue, code: 0, userInfo: [NSLocalizedDescriptionKey: "text background color is required"])
        }
        self.colorBg = UIColor(hex: color) ?? UIColor.clear
        if textBackground.keys.contains("cornerRadius") {
            guard let cornerRadius = textBackground["cornerRadius"] as? [AnyHashable: Any] else {
                throw NSError(domain: ErrorDomainEnum.PARAMS_INVALID.rawValue, code: 0, userInfo: [NSLocalizedDescriptionKey: "cornerRadius is invalid"])
            }
            self.cornerRadius = try CornerRadius(dicOpts: cornerRadius)
        }
    }
}
